src/containers/search_bar.js

```diff
@@ -1,29 +1,60 @@
 import React, { Component } from 'react';
+import _ from 'lodash';
 import { connect } from 'react-redux';
 import { searchMovies } from '../actions';
+import { SearchLoader } from 'components/loader';
 import SearchedMovies from '../components/searched_movies';
 
 class SearchBar extends Component {
 
   constructor(props) {
     super(props);
 
-    this.state = { inputTerm:'' }
-    this.inputSearchTerm = this.inputSearchTerm.bind(this);
+    this.state = { 
+      inputTerm:'',
+      searchLoader: false,
+      pageLoader: false,
+      showMovieList: false
+    }
+    this.searchBar = React.createRef();
+    this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
+    this.inputSearchTerm = _.debounce( this.inputSearchTerm.bind(this), 250 );
+    this.showMovieList = this.showMovieList.bind(this);
+    this.handleSearchTermFocus= this.handleSearchTermFocus.bind(this);
+  };
+
+  handleSearchTermChange(event) {
+    event.persist();
+    const value = event.target.value;
+    this.setState({ searchLoader: true, inputTerm: value })
+    this.inputSearchTerm(value)
   }
  
- 
-  
-  inputSearchTerm(e) {
-    this.setState({inputTerm: e.target.value}, () => {
-      // Condition for not rendering movies if the search string is ''
-      this.state.inputTerm ? 
-      this.props.searchMovies({query_term: this.state.inputTerm}
-      ) : null
-       
-    });
+  inputSearchTerm(value) {
+    // Condition for not rendering movies if the search string is ''
+    value.length >= 1 ? this.props.searchMovies( { query_term: value} )
+    .then(() => this.setState({ searchLoader: false, showMovieList: true })) : this.setState({ searchLoader: false, showMovieList: false })
+  }
+
+  showMovieList(event) {
+    this.searchBar.current.contains(event.target) ?  this.setState({ showMovieList: true })
+     : this.setState({ showMovieList: false })
   }
 
+  handleSearchTermFocus(event) {
+    event.stopPropagation();
+    this.setState({ showMovieList: true })
+  }
+
+  componentDidMount() {
```

Comment:
You are attaching an event listener on the document. Also make sure to remove the same on componentWillUnmount.